### PR TITLE
search for libretro system libraries

### PR DESCRIPTION
**Pull request by Rechi**
This allows using libretro system libs.

### CODE REVIEW

File: templates/addon/CMakeLists.txt.j2 [hunk 1]
```diff
@@ -4,7 +4,7 @@ project({{ game.addon }})
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR})
 
 find_package(Kodi REQUIRED)
-find_package(libretro-{{ game.name }} REQUIRED)
+find_library({{ game.name|upper }}_LIB NAMES {{ library.soname }}${CMAKE_SHARED_LIBRARY_SUFFIX} PATH_SUFFIXES libretro)
```
Comment (Rechi) [inline templates/addon/CMakeLists.txt.j2]: If wanted I can make this a correct find_package module.

**Comment by fetzerch** [comment]
I think we do have to make sure that the system wide installed version of the libretro core matches exactly the version that we're trying to package (git sha). The script extracts version information from the compiled libretro core as well as supported file system extensions and the setting the core supports.

If you just take whatever system version is installed, then kodi will show that game.libretro.x is installed in version 2.1.0 (with the settings that this version supports), while the actual provided version is something completely different.

Might be I'm overlooking something, but what's your idea of matching addon and libretro core version?

**Comment by Rechi** [comment]
I have two solutions in mind
- check if version of the found lib matches the expected one
- set the lib version as addon version (and other info)

Which one would you prefer?

**Comment by fetzerch** [comment]
I'd prefer the first one. Could we fallback to compiling it if there's no matching version installed on the system?

**Comment by Rechi** [comment]
Either you want to compile with depends or system libs.

**Comment by Rechi** [comment]
With option one how would you handle cores which aren't versioned properly?
Option 2 would allow to read all the info at compile time.

**Comment by fetzerch** [comment]
Most of them are not versioned properly. The version number typically reflects the version of the emulator. But all changes to make it a libretro core, bug fixes, ... is not reflected in the version number. The most reliable way would be the git sha. Packages from [official libretro repos](https://launchpad.net/~libretro/+archive/ubuntu/stable) do contain the git sha in the version number. Can we read that somehow?

Option 2 would mean to reimplement quite a bunch of stuff from this project in cmake? reading versions, generating settings, ...

Always compiling the game addons is not an option? The libretro cores are somewhat different to other depends. It's the main addon code we'll be running. Not just for example an xml-library that doesn't impact the functionality or feature set too much.

**Comment by garbear** [comment]
Libretro version numbers are cosmetic, like the emulator name. They usually reflect the version of a past fork. They shouldn't be used in any versioning logic.

**Comment by Rechi** [comment]
Detecting the version is simple, but if you care about creating debian packages, the control file should set the correct libretro version requirements.
I initially opened this PR to get rid of writing a config file for people who know what they are doing (e.g. [LibreELEC](https://github.com/LibreELEC/LibreELEC.tv/blob/95e37a1/packages/emulation/libretro-2048/package.mk)). There is no version check and relying the config file doesn't assert that correct version is used too.

> Always compiling the game addons is not an option?

In the current way they are pulled in, no, they are dependencies. If you always want to build them, the source code has to be imported into the game.libretro.* repos.

**Comment by fetzerch** [comment]
@Rechi you're right, I'll merge this and update the addons. Ensuring the right version is used is an extra step.

**Pull request merged**